pytest_arraydiff/plugin.py

```diff
@@ -310,7 +310,7 @@ def pytest_runtest_call(self, item):
         # Find test name to use as plot name
         filename = compare.kwargs.get('filename', None)
         if filename is None:
-            filename = item.name + '.' + extension
+            filename = item.function + '.' + extension
             if not single_reference:
                 filename = filename.replace('[', '_').replace(']', '_')
                 filename = filename.replace('_.' + extension, '.' + extension)
```

Comment:
I think this should work:
```py
        if filename is None:
            if single_reference:
                filename = item.function + '.' + extension
            else:
                filename = item.name + '.' + extension
                filename = filename.replace('[', '_').replace(']', '_')
                filename = filename.replace('_.' + extension, '.' + extension)
```

Reply:
Ah, thanks! I updated the patch and added you as co-author of the commit.